modules/programs/lazygit.nix

```diff
@@ -0,0 +1,54 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.lazygit;
+
+  yamlFormat = pkgs.formats.yaml { };
+
+  inherit (pkgs.stdenv.hostPlatform) isDarwin;
+
+in {
+  meta.maintainers = [ maintainers.kalhauge ];
+
+  options.programs.lazygit = {
+    enable = mkEnableOption "simple terminal UI for git commands";
+
+    settings = mkOption {
+      type = yamlFormat.type;
+      default = { };
+      example = {
+        gui.theme = {
+          lightTheme = true;
+          activeBorderColor = [ "blue" "bold" ];
+          inactiveBorderColor = [ "black" ];
+          selectedLineBgColor = [ "default" ];
+        };
+      };
+      description = ''
+        Configuration written to
+        <filename>~/.config/lazygit/config.yml</filename> on linux 
+        or <filename>~/Library/Application Support/lazygit/config.yml</filename> on darwin. See
+        <link xlink:href="https://github.com/jesseduffield/lazygit/blob/master/docs/Config.md"/>
+        for supported values.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    home.packages = [ pkgs.lazygit ];
+
+    home.file."Library/Application Support/lazygit/config.yml" =
+      mkIf (cfg.settings != { } && isDarwin) {
```

Comment:
Does it matter if `cfg.settings = { }`? I think that would just create an empty file, but is that a bad thing?

Reply:
I'm mostly trying to follow the precedence from lsd:
https://github.com/nix-community/home-manager/blob/6aa6556bcab6dc0f6398b4daa8404d788fd7a6a2/modules/programs/lsd.nix#L58

Here is the pros and cons as I see them:

- Pros: Home-manager will reduce the number of symbolic links / pollution made in peoples home-directories. If people already have a file configuration on a computer, and try to install this program with the no settings it will not refuse to install.
- Cons: You might not be warned that a configuration already exists, when this module is given no settings, which might result in inconsistent configurations.

Reply:
I might go for the more consistent configuration, in which case the following places need an update:

```text
$ git grep 'cfg.settings != { }'
modules/misc/dconf.nix:  config = mkIf (cfg.enable && cfg.settings != { }) {
modules/programs/alacritty.nix:      xdg.configFile."alacritty/alacritty.yml" = mkIf (cfg.settings != { }) {
modules/programs/beets.nix:          cfg.settings != { };
modules/programs/lazygit.nix:      mkIf (cfg.settings != { } && isDarwin) {
modules/programs/lazygit.nix:      mkIf (cfg.settings != { } && !isDarwin) {
modules/programs/lsd.nix:    xdg.configFile."lsd/config.yaml" = mkIf (cfg.settings != { }) {
modules/programs/noti.nix:      mkIf (cfg.settings != { }) { text = generators.toYAML { } cfg.settings; };
modules/programs/nushell.nix:    xdg.configFile."nu/config.toml" = mkIf (cfg.settings != { }) {
modules/programs/powerline-go.nix:  otherSettingsArgument = optionalString (cfg.settings != { })
modules/programs/starship.nix:    xdg.configFile."starship.toml" = mkIf (cfg.settings != { }) {
modules/services/dunst.nix:    (mkIf (cfg.settings != { }) {
```

Reply:
I really have no preference/say in the matter. Do whatever you feel is best :)

Reply:
I'll leave it as is... maybe we can open an issue on this if it becomes a problem.